components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts

```diff
@@ -25,6 +25,68 @@ const getWindowId = (id: number) => {
   return `id_${id}`
 }
 
+function pollTwitchPage (tab: chrome.tabs.Tab, tabId: number, publisherBlob:string): void {
+  let url = tab.url
+  // use an interval here to monitor when the DOM has finished
+  // generating. clear after the data is present.
+  // Check every second no more than 'limit' times
+  // clear the interval if panel closes
+  const markupMatch = '<figure class=\"tw-avatar tw-avatar--size-36\">' +
+                      '<div class=\"tw-border-radius-medium tw-overflow-hidden\">' +
+                      '<img class=\"tw-avatar__img tw-image\" alt=\"'
+  let obtained = false
+  // run once to see if we already have it
+  chrome.tabs.executeScript(tabId, {
+    code: 'var outerHtml = document.body.outerHTML; outerHtml'
+  }, function (result: string[]) {
+    if (result[0].includes(markupMatch)) {
+      obtained = true
+      publisherBlob = result[0]
+      if (url) {
+        chrome.braveRewards.onTwitchInfo(publisherBlob, url)
+      }
+      if (tab && tab.windowId && tab.url && tab.favIconUrl) {
+        chrome.braveRewards.getTwitchPublisherData(tab.windowId, tab.url, tab.favIconUrl || '', publisherBlob)
+      }
+    }
+  })
+
+  // if not obtained, then poll page for reasonable loading time
+  if (!obtained) {
```

Comment:
this is always going to be false the first time because https://github.com/brave/brave-core/pull/888#diff-fc21efee53a5e1ba98eb186ef1e788a8R39 is async. can you just delete https://github.com/brave/brave-core/pull/888#diff-fc21efee53a5e1ba98eb186ef1e788a8R39 and change this to `while (!obtained && itr < limit) {...}` (moving limit out of the while block). that makes it easier to understand avoids code repetition.

Reply:
This is always expected to be false when page initially loads. I added the top `executeScript` for when all of the generated DOM has loaded and the panel has already been displayed, the user closes the panel and opens it again.
Basically it's for use after panel has been displayed on publisher page, otherwise it delays showing publisher and shows 'twitch.tv' for `interval` seconds.

Reply:
@jasonrsadler right - what i'm saying is that the top call to executeScript seems unnecessary since it sets `obtained` after this `if` condition has been checked, so it can be refactored way

Reply:
Yep. Agree. I edited my previous response.

Reply:
Fixed.